src/submission/tests/test_views.py

```diff
@@ -0,0 +1,239 @@
+__copyright__ = "Copyright 2025 Birkbeck, University of London"
+__author__ = "Open Library of Humanities"
+__license__ = "AGPL v3"
+__maintainer__ = "Open Library of Humanities"
+
+from django.test import TestCase, override_settings
+from unittest.mock import patch
+
+from submission import forms, logic, models, views
+from utils import setting_handler
+from utils.testing import helpers
+
+
+class TestSubmitAuthors(TestCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.press = helpers.create_press()
+        cls.journal_one, cls.journal_two = helpers.create_journals()
+        cls.kathleen = helpers.create_user(
+            'haogjlbdmax7b6tnc6qa@example.org',
+            roles=['author'],
+            journal=cls.journal_one,
+            first_name='Kathleen',
+            last_name='Booth',
+            is_active=True,
+        )
+        cls.eliot = helpers.create_user(
+            'soqdbzkiz7movjtix6hr@example.org',
+            roles=['author'],
+            journal=cls.journal_one,
+            first_name='T.',
+            middle_name='S.',
+            last_name='Eliot',
+        )
+        cls.article = helpers.create_article(
+            cls.journal_one,
+            owner=cls.kathleen,
+            current_step=2,
+        )
+        setting_handler.save_setting(
+            'general',
+            'limit_access_to_submission',
+            cls.journal_one,
+            '',
+        )
+        setting_handler.save_setting(
+            'general',
+            'disable_journal_submission',
+            cls.journal_one,
+            '',
+        )
+        logic.add_user_as_author(cls.kathleen, cls.article)
+
+    @override_settings(URL_CONFIG='domain')
+    def test_submit_authors_get(self):
+        self.client.force_login(self.kathleen)
+        response = self.client.get(
+            f'/submit/{self.article.pk}/authors/',
```

Comment:
Why aren't we reversing the URL here? If the URL pattern changes the test will break.

Reply:
The first few view tests I wrote for Janeway were based on the Django docs, which do not reverse URLs but write them out.

https://docs.djangoproject.com/en/4.2/topics/testing/tools/#overview-and-a-quick-example

Later I noticed this was different from the way they've been written in Janeway before. At this point I've added 50+ tests with literal URLs. Shall I reverse them?